README.md

```diff
@@ -0,0 +1,64 @@
+# TH2 'Bill of Materials' (BOM) project
+
+This project contains list of dependencies with their versions that are compatible and can be used with each other.
+
+**NOTE: the BOM can be used only for Java and Java based projects. All, dependencies specified in it are Java libraries**
+
+It provides users who interact with the TH2 (weather he or she writes a script or a new component) with ability to
+specify dependencies that are required for development in TH2 environment without defining a version for those dependencies.
+The version will be taken from **the BOM** and it guarantees that different TH2 dependencies _will be compatible with each other_.
+Also, it guarantees that if you use some libraries that are also used by TH2 you _won't use incompatible libraries' versions_.
+
+More information about what is 'BOM' and what is its purpose you can find [here](https://en.wikipedia.org/wiki/Software_bill_of_materials).
+
+## When you need to use BOM?
+
+If you are developing script or a new component for TH2 you need to use certain libraries for integration with th2-schema and other components.
+You need to know what version of each component to use and what versions are compatible.
+Also, you need to update versions sometimes.
+In this case, you need to know which components versions to upgrade and if they are compatible with other dependencies.
+
+Here BOM comes in. It allows you to specify only the BOM's version in your dependencies.
+The versions from other components will be taken from BOM (if they are present there) and you only need to specify which dependencies you need.
+
+If you need to upgrade to newer components you will only need to update the BOM's version and versions of your dependencies will be upgraded too.
+And they will be compatible with each other.
+
+## BOM's versioning
+
+The BOM's uses [semantic version](https://semver.org/). That means you can rely on its version to keep compatibility and do not break your code.
+
+If you upgrade the **PATCH/MAINTENANCE** of BOM your code still will compile and keep working with components
```

Comment:
If you upgrade the **PATCH/MAINTENANCE** of BOM **,** your code **will still** compile and keep working with **the** components